Modules/Quotas/src/Quotas.Application/Identities/Queries/GetQuotasForIdentity/Handler.cs

```diff
@@ -0,0 +1,85 @@
+﻿using Backbone.BuildingBlocks.Application.Pagination;
+using Backbone.DevelopmentKit.Identity.ValueObjects;
+using Backbone.Modules.Quotas.Application.DTOs;
+using Backbone.Modules.Quotas.Application.Infrastructure.Persistence.Repository;
+using Backbone.Modules.Quotas.Domain.Aggregates.Identities;
+using Backbone.Modules.Quotas.Domain.Aggregates.Metrics;
+using Backbone.Modules.Quotas.Domain.Metrics;
+using MediatR;
+using Microsoft.Extensions.Caching.Memory;
+
+namespace Backbone.Modules.Quotas.Application.Identities.Queries.GetQuotasForIdentity;
+
+public class Handler : IRequestHandler<ListQuotasForIdentityQuery, ListQuotasForIdentityResponse>
+{
+    private readonly IMemoryCache _cache;
+
+    private readonly IIdentitiesRepository _identitiesRepository;
+    private readonly MetricCalculatorFactory _metricCalculatorFactory;
+
+    public Handler(IMemoryCache cache, IIdentitiesRepository identitiesRepository, MetricCalculatorFactory metricCalculatorFactory)
+    {
+        _cache = cache;
+        _identitiesRepository = identitiesRepository;
+        _metricCalculatorFactory = metricCalculatorFactory;
+    }
+
+    public async Task<ListQuotasForIdentityResponse> Handle(ListQuotasForIdentityQuery request, CancellationToken cancellationToken)
+    {
+        var identityList = await _identitiesRepository.FindByAddresses(new List<string> { request.Address }.AsReadOnly(), cancellationToken, true);
+        var identity = identityList.First();
```

Comment:
This method throws an exception if the list is empty. This results in an http status 500.